### PR TITLE
Added tree view context menu commands

### PR DESCRIPTION
**Pull request by mrveress**
@jonathanrico, sometimes, mr Barry Allen is too lazy. He don't want to open each file in text editor for separate deploy/retrieve. 
So, this functionality is migh be very helpful for mr Allen and for everyone else :)

### CODE REVIEW

File: lib/build.coffee [hunk 1]
```diff
@@ -1,6 +1,7 @@
 child_process = require 'child_process'
 fs = require 'fs'
 qs = require 'querystring'
+pathModule = require 'path'
```
Comment (mrveress) [inline lib/build.coffee:4]: This is for BaseName functionality

File: lib/build.coffee [hunk 2]
```diff
@@ -96,20 +99,30 @@ module.exports =
...
+      if(cmdtype == 'editor' && atom.workspace.getActiveTextEditor()?.buffer?.file?)
         path = atom.workspace.getActiveTextEditor().buffer.file.path
+      else if(cmdtype == 'treeview' && atom.packages.getActivePackage('tree-view')?.mainModule.createView().selectedPaths()?)
+        path = atom.packages.getActivePackage('tree-view').mainModule.createView().selectedPaths()[0]
+      if(path)
+        isWin = /^win/.test(process.platform)
         projectPath = if isWin then @root+'\\src\\' else @root+'/src/'
         if(path.startsWith(projectPath))
-          fileBaseName = atom.workspace.getActiveTextEditor().buffer.file.getBaseName()
+          fileBaseName = pathModule.basename(path)
```
Comment (mrveress) [inline lib/build.coffee:125]: Here we get the base name without recheck active text editor

File: lib/build.coffee [hunk 3]
```diff
@@ -96,20 +99,30 @@ module.exports =
...
+  processSingleFile: (optype, cmdtype) ->
     if(@isDeployRunning())
       clearTimeout @finishedTimer
-      if(atom.workspace.getActiveTextEditor().buffer?.file?)
-        isWin = /^win/.test(process.platform)
+      path = null
+      if(cmdtype == 'editor' && atom.workspace.getActiveTextEditor()?.buffer?.file?)
         path = atom.workspace.getActiveTextEditor().buffer.file.path
+      else if(cmdtype == 'treeview' && atom.packages.getActivePackage('tree-view')?.mainModule.createView().selectedPaths()?)
+        path = atom.packages.getActivePackage('tree-view').mainModule.createView().selectedPaths()[0]
```
Comment (mrveress) [inline lib/build.coffee:120]: Here we get path from editor/tree-vew

File: package.json [hunk 4]
```diff
@@ -7,13 +7,15 @@
...
+      "build:sf-deploy-file-treeview",
       "build:sf-deploy",
       "build:sf-deploy-static-res",
       "build:sf-deploy-apex",
       "build:sf-deploy-visualforce",
       "build:sf-abort",
       "build:sf-retrieve-unpackaged",
-      "build:sf-retrieve-unpackaged-file"
+      "build:sf-retrieve-unpackaged-file",
+      "build:sf-retrieve-unpackaged-file-treeview"
```
Comment (mrveress) [inline package.json:18]: Activate module when we call this commands too

**Comment by jonathanrico** [comment]
this is awesome! thanks @mrveress !

**Pull request merged**